t/test01.t

```diff
@@ -68,57 +86,72 @@ my $frontend_params_1 = {
 sub run_zonemaster_test_with_backend_API {
     my ($test_id) = @_;
 
-    my $hash_id;
-    ok( ($hash_id = $engine->start_domain_test( $frontend_params_1)) && $hash_id , "API start_domain_test OK/test_id=$test_id" );
-    ok( scalar( $engine->{db}->dbh->selectrow_array( qq/SELECT id FROM test_results WHERE id=$test_id/ ) ) == $test_id );
+    my $hash_id = $engine->start_domain_test( $frontend_params_1 );
+    ok( $hash_id, "API start_domain_test OK" );
+    is( length($hash_id), 16, "Test has a 16 characters length hash ID (hash_id=$hash_id)" );
+    is( scalar( $engine->{db}->dbh->selectrow_array( qq/SELECT id FROM test_results WHERE id=$test_id/ ) ), $test_id , 'API start_domain_test -> Test inserted in the DB' );
 
     # test test_progress API
-    ok( $engine->test_progress( $hash_id ) == 0 );
-
-    use_ok( 'Zonemaster::Backend::Config' );
-
-    use_ok( 'Zonemaster::Backend::TestAgent' );
+    is( $engine->test_progress( $hash_id ), 0 , 'API test_progress -> OK');
 
     if ( not $ENV{ZONEMASTER_RECORD} ) {
         Zonemaster::Engine->preload_cache( $datafile );
         Zonemaster::Engine->profile->set( q{no_network}, 1 );
     }
-    Zonemaster::Backend::TestAgent->new( { dbtype => "SQLite", config => $config } )->run( $hash_id );
+
+    use_ok( 'Zonemaster::Backend::TestAgent' );
+    Zonemaster::Backend::TestAgent->new( { dbtype => "$db_backend", config => $config } )->run( $hash_id );
 
     Zonemaster::Backend::TestAgent->reset() unless ( $ENV{ZONEMASTER_RECORD} );
 
-    ok( $engine->test_progress( $hash_id ) > 0 );
+    sleep( 5 ) unless ( $ENV{ZONEMASTER_RECORD} );
+    cmp_ok( $engine->test_progress( $hash_id ), '>',  0 , 'API test_progress -> Test started');
 
     foreach my $i ( 1 .. 12 ) {
+        sleep( 5 ) unless ( $ENV{ZONEMASTER_RECORD} );
```

Comment:
I tried to think about this sleep call too. If we're recording we'll poll twelve times back-to-back, and if we're using a preloaded cache we poll for a minute. Shouldn't it be the other way around?

Reply:
Removed in 741b75b